src/wallet/spend.cpp

```diff
@@ -940,7 +940,9 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
 
     // The only time that fee_needed should be less than the amount available for fees is when
     // we are subtracting the fee from the outputs. If this occurs at any other time, it is a bug.
-    assert(coin_selection_params.m_subtract_fee_outputs || fee_needed <= nFeeRet);
+    if (!coin_selection_params.m_subtract_fee_outputs && fee_needed > nFeeRet) {
+        return util::Error{strprintf(_("Internal bug detected: Fee needed > fee required. Please report this issue to %s"), PACKAGE_BUGREPORT)};
```

Comment:
nit: Is there any value in translating an internal bug report, which ideally should never happen?

Also, it could make sense to make it usable in other places as well by defining a macro for it. This could allow to auto-attach more info to the bug report, such as the version and source code location.


Example:

```diff
diff --git a/src/util/check.cpp b/src/util/check.cpp
index e50d6087d0..b17d48bfb1 100644
--- a/src/util/check.cpp
+++ b/src/util/check.cpp
@@ -14,10 +14,13 @@
 #include <cstdlib>
 #include <string>
 
+std::string StrFormatInternalBug(const char* msg, const char* file, int line, const char* func)
+{
+    return strprintf("Internal bug detected: \"%s\"\n%s:%d (%s)\nPlease report this issue here: %s\n", msg, file, line, func, PACKAGE_BUGREPORT);
+}
 
 NonFatalCheckError::NonFatalCheckError(const char* msg, const char* file, int line, const char* func)
-    : std::runtime_error{
-          strprintf("Internal bug detected: \"%s\"\n%s:%d (%s)\nPlease report this issue here: %s\n", msg, file, line, func, PACKAGE_BUGREPORT)}
+    : std::runtime_error{StrFromatInternalBug(msg, file, line, func)}
 {
 }
 
diff --git a/src/util/check.h b/src/util/check.h
index b6c03bed2a..b791944502 100644
--- a/src/util/check.h
+++ b/src/util/check.h
@@ -10,12 +10,16 @@
 #include <stdexcept>
 #include <utility>
 
+std::string StrFormatInternalBug(const char* msg, const char* file, int line, const char* func);
+
 class NonFatalCheckError : public std::runtime_error
 {
 public:
     NonFatalCheckError(const char* msg, const char* file, int line, const char* func);
 };
 
+#define STR_INTERNAL_BUG(msg) StrFormatInternalBug((msg), __FILE__, __LINE__, __func__)
+
 /** Helper for CHECK_NONFATAL() */
 template <typename T>
 T&& inline_check_non_fatal(LIFETIMEBOUND T&& val, const char* file, int line, const char* func, const char* assertion)

Reply:
Added this diff as a new commit, and changed to use it `STR_INTERNAL_BUG`.